packages/dds/sequence/src/test/intervalRevertibles.fuzz.spec.ts

```diff
@@ -0,0 +1,376 @@
+		const { start, end } = inclusiveRange(state);
+		return {
+			type: "changeInterval",
+			start: state.random.integer(0, 5) === 5 ? undefined : start,
+			end: state.random.integer(0, 5) === 5 ? undefined : end,
+			...interval(state),
+		};
+	}
+
+	async function changeProperties(state: ClientOpState): Promise<ChangeProperties> {
+		return {
+			type: "changeProperties",
+			...interval(state),
+			properties: propertySet(state),
+		};
+	}
+
+	async function revertSharedStringRevertibles(
+		state: ClientOpState,
+	): Promise<RevertSharedStringRevertibles> {
+		return {
+			type: "revertSharedStringRevertibles",
+			...interval(state),
+		};
+	}
+
+	const hasAnInterval = ({ channel }: ClientOpState): boolean =>
+		Array.from(channel.getIntervalCollectionLabels()).some((label) => {
+			const collection = channel.getIntervalCollection(label);
+			return isNonEmpty(collection);
+		});
+
+	const lengthSatisfies =
+		(criteria: (length: number) => boolean): AcceptanceCondition<ClientOpState> =>
+		({ channel }) =>
+			criteria(channel.getLength());
+	const hasNonzeroLength = lengthSatisfies((length) => length > 0);
+	const isShorterThanMaxLength = lengthSatisfies((length) => length < options.maxStringLength);
+
+	const hasNotTooManyIntervals: AcceptanceCondition<ClientOpState> = ({ channel }) => {
+		let intervalCount = 0;
+		for (const label of channel.getIntervalCollectionLabels()) {
+			for (const _ of channel.getIntervalCollection(label)) {
+				intervalCount++;
+				if (intervalCount >= options.maxIntervals) {
+					return false;
+				}
+			}
+		}
+		return true;
+	};
+
+	const all =
+		<T>(...clauses: AcceptanceCondition<T>[]): AcceptanceCondition<T> =>
+		(t: T) =>
+			clauses.reduce<boolean>((prev, cond) => prev && cond(t), true);
+
+	return createWeightedGenerator<RevertOperation, ClientOpState>([
+		[addText, 2, isShorterThanMaxLength],
+		[removeRange, 1, hasNonzeroLength],
+		// [addInterval, 0, all(hasNotTooManyIntervals, hasNonzeroLength)],
+		[addInterval, 2, all(hasNotTooManyIntervals, hasNonzeroLength)],
+		// [deleteInterval, 2, hasAnInterval],
+		// [changeInterval, 2, all(hasAnInterval, hasNonzeroLength)],
+		[changeProperties, 2, hasAnInterval],
+		// don't know what acceptance condition i should be using
+		[revertSharedStringRevertibles, 2, hasAnInterval],
+	]);
+}
+
+describe.skip("IntervalCollection fuzz testing", () => {
+	const model: DDSFuzzModel<SharedStringFactory, RevertOperation, FuzzTestState> = {
+		workloadName: "interval collection with revertibles",
+		generatorFactory: () => take(100, makeOperationGenerator()),
+		reducer:
+			// makeReducer supports a param for logging output which tracks the provided intervalId over time:
+			// { intervalId: "00000000-0000-0000-0000-000000000000", clientIds: ["A", "B", "C"] }
+			makeRevertibleReducer(),
+		validateConsistency: assertEquivalentSharedStrings,
+		factory: new SharedStringFactory(),
+	};
+
+	createDDSFuzzSuite(
+		model,
+		// optionsWithEmitter,
```

Comment:
Passing in the previously created object instead of the inline object here causes only one test to run instead of all 100...not sure why this is happening, but they all run when I just add the emitter to the options here.